v7/flag_overrides.go

```diff
@@ -0,0 +1,136 @@
+// feature flags downloaded from the ConfigCat CDN with local values.
+//
+// With Values, you can set up the SDK to load your feature flag overrides from a map.
+//
+// With FilePath, you can set up the SDK to load your feature flag overrides from a JSON file.
+type FlagOverrides struct {
+	// Behaviour describes how the overrides should behave. Default is LocalOnly.
+	Behaviour OverrideBehaviour
+
+	// Values is a map that contains the overrides.
+	Values map[string]interface{}
+
+	// FilePath is the path to a JSON file that contains the overrides.
+	FilePath string
+
+	entries map[string]*wireconfig.Entry
+}
+
+func (f *FlagOverrides) isValid() bool {
+	return f.Values != nil || f.FilePath != ""
+}
+
+func (f *FlagOverrides) preLoad(logger *leveledLogger) {
+	if f.Behaviour != LocalOnly && f.Behaviour != LocalOverRemote && f.Behaviour != RemoteOverLocal {
+		logger.Errorf("flag overrides behaviour configuration is invalid. 'Behavior' is %v.", f.Behaviour)
+		return
+	}
+	if !f.isValid() {
+		logger.Errorf("flag overrides configuration is invalid. 'Values' or 'FilePath' must be set.")
+		return
+	}
+
+	f.entries = f.loadEntries(logger)
+	f.fixEntries()
+}
+
+func (f *FlagOverrides) loadEntries(logger *leveledLogger) map[string]*wireconfig.Entry {
+	if f.Values != nil {
+		entries := make(map[string]*wireconfig.Entry, len(f.Values))
+		for key, value := range f.Values {
+			entries[key] = &wireconfig.Entry{
+				Value:       value,
+				VariationID: "",
+			}
+		}
+		return entries
+	}
+	return f.loadEntriesFromFile(logger)
+}
+
+func (f *FlagOverrides) loadEntriesFromFile(logger *leveledLogger) map[string]*wireconfig.Entry {
+	data, err := ioutil.ReadFile(f.FilePath)
+	if err != nil {
+		logger.Errorf("unable to read local JSON file: %v", err)
+		return nil
+	}
+	var simplified wireconfig.SimplifiedConfig
+	reader := bytes.NewReader(data)
+	decoder := json.NewDecoder(reader)
+	decoder.UseNumber()
+	if err := decoder.Decode(&simplified); err == nil && simplified.Flags != nil {
+		entries := make(map[string]*wireconfig.Entry, len(simplified.Flags))
+		for key, value := range simplified.Flags {
+			entries[key] = &wireconfig.Entry{
+				Value:       value,
+				VariationID: "",
+			}
+		}
+		return entries
+	}
+	var root wireconfig.RootNode
+	if _, err = reader.Seek(0, io.SeekStart); err != nil {
+		logger.Errorf("error during reading local JSON file: %v", err)
+		return nil
+	}
+	if err := decoder.Decode(&root); err != nil {
+		logger.Errorf("error during reading local JSON file: %v", err)
+		return nil
+	}
+	return root.Entries
+}
+
+func (f *FlagOverrides) fixEntries() {
+	if f.entries == nil {
+		return
+	}
+	for _, entry := range f.entries {
+		if b, ok := entry.Value.(bool); ok {
+			entry.Value = b
+			entry.Type = wireconfig.BoolEntry
+		} else if s, ok := entry.Value.(string); ok {
+			entry.Value = s
+			entry.Type = wireconfig.StringEntry
+		} else if n, ok := entry.Value.(json.Number); ok {
+			if i, err := n.Int64(); err == nil {
```

Comment:
This makes me uneasy. JSON as a file format doesn't distinguish between int and float, so if this JSON has been through any kind of translator, any trailing `.0` might well be stripped off, which would cause a float value to be interpreted as int here.

One alternative might be to allow an int flag to read a float value. Not entirely ideal but perhaps better than this, which seems like a potential footgun to me.

Reply:
SGTM in general, but I think it could be a little simpler if it worked the other way around.